hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/DecommissionOMSubcommand.java

```diff
@@ -0,0 +1,206 @@
+      description = "Host name/address of the OM to be decommissioned.",
+      required = true)
+  private String hostname;
+
+  private InetAddress hostInetAddress;
+
+  @CommandLine.Option(
+      names = {"--force"},
+      description = "This option will skip checking whether OM configs " +
+          "have been updated with the decommissioned node added to " +
+          "ozone.om.decommissioned.nodes config in ozone-site.xml."
+  )
+  private boolean force;
+
+  private OzoneConfiguration ozoneConf;
+  private UserGroupInformation user;
+
+  @Override
+  public Void call() throws IOException {
+    ozoneConf = parent.getParent().getOzoneConf();
+    user = parent.getParent().getUser();
+
+    verifyNodeIdAndHostAddress();
+    if (!force) {
+      verifyConfigUpdatedOnAllOMs();
+    }
+
+    // Proceed with decommissioning the OM by contacting the current OM
+    // leader.
+    try (OMAdminProtocolClientSideImpl omAdminProtocolClient =
+             OMAdminProtocolClientSideImpl.createProxyForOMHA(ozoneConf, user,
+                 omServiceId)) {
+      OMNodeDetails decommNodeDetails = new OMNodeDetails.Builder()
+          .setOMNodeId(decommNodeId)
+          .setHostAddress(hostInetAddress.getHostAddress())
+          .build();
+      omAdminProtocolClient.decommission(decommNodeDetails);
+
+      System.out.println("Successfully decommissioned OM " + decommNodeId);
+    } catch (IOException e) {
+      System.out.println("Failed to decommission OM " + decommNodeId);
+      throw e;
+    }
+    return null;
+  }
+
+  /**
+   * Verify that the provided nodeId and host address correspond to the same
+   * OM in the configs.
+   */
+  private void verifyNodeIdAndHostAddress() throws IOException {
+    String rpcAddrKey = ConfUtils.addKeySuffixes(OZONE_OM_ADDRESS_KEY,
+        omServiceId, decommNodeId);
+    String rpcAddrStr = OmUtils.getOmRpcAddress(ozoneConf, rpcAddrKey);
+    if (rpcAddrStr == null || rpcAddrStr.isEmpty()) {
+      throw new IOException("There is no OM corresponding to " + decommNodeId
+          + "in the configuration.");
+    }
+
+    hostInetAddress = InetAddress.getByName(hostname);
+    InetAddress rpcAddressFromConfig = InetAddress.getByName(
+        rpcAddrStr.split(":")[0]);
+
+    if (!hostInetAddress.equals(rpcAddressFromConfig)) {
+      throw new IOException("OM " + decommNodeId + "'s host address in " +
+          "config - " + rpcAddressFromConfig.getHostAddress() + " does not " +
+          "match the provided host address " + hostInetAddress);
+    }
+  }
+
+  /**
+   * Verify that the to be decommissioned node is added to the
+   * OZONE_OM_DECOMMISSIONED_NODES_KEY.<SERVICE_ID> config in ozone-site.xml
+   * of all OMs.
+   */
+  private void verifyConfigUpdatedOnAllOMs() throws IOException {
+    String decommNodesKey = ConfUtils.addKeySuffixes(
+        OZONE_OM_DECOMMISSIONED_NODES_KEY, omServiceId);
+    Collection<String> decommNodes = ozoneConf.getTrimmedStringCollection(
```

Comment:
This means before decomission, the config needs to be updated on all client/OMs. Is it correct?

Reply:
Yes, would have to update config on all OMs atleast before decommissioning.